src/endpoints/assets/v1.ts

```diff
@@ -0,0 +1,618 @@
+            upper_bound?: string
+            before?: number
+            after?: number
+            page?: number
+            limit?: number
+            order?: 'asc' | 'desc'
+            sort?: 'asset_id' | 'minted' | 'updated' | 'transferred' | 'template_mint' | 'name'
+        },
+        extra_options?: {[key: string]: string}
+    ) {
+        const queryParts = {}
+
+        for (const [key, value] of Object.entries(options || {})) {
+            queryParts[key] = value as string
+        }
+
+        for (const [key, value] of Object.entries(extra_options || {})) {
+            queryParts[key] = value
+        }
+
+        const queryParams = Object.keys(queryParts).length
+            ? '?' + new URLSearchParams(queryParts).toString()
+            : ''
+
+        return this.client.call({
+            path: `/atomicassets/v1/assets${queryParams}`,
+            method: 'GET',
+            responseType: Assets.GetAssetsResponse,
+        })
+    }
+
+    async get_asset(asset_id: UInt64Type) {
+        return this.client.call({
+            path: `/atomicassets/v1/assets/${asset_id}`,
+            method: 'GET',
+            responseType: Assets.GetAssetResponse,
+        })
+    }
+
+    async get_asset_stats(asset_id: UInt64Type) {
+        return this.client.call({
+            path: `/atomicassets/v1/assets/${asset_id}/stats`,
+            method: 'GET',
+            responseType: Assets.GetAssetStatsResponse,
+        })
+    }
+
+    async get_asset_logs(
+        asset_id: UInt64Type,
+        options?: {
+            page?: number
+            limit?: number
+            order?: 'asc' | 'desc'
+            action_whitelist?: ActionType[]
+            action_blacklist?: ActionType[]
+        }
+    ) {
+        const queryParts = {}
+
+        for (const [key, value] of Object.entries(options || {})) {
+            queryParts[key] = value as string
+        }
+
+        const queryParams = Object.keys(queryParts).length
+            ? '?' + new URLSearchParams(queryParts).toString()
+            : ''
+
+        return this.client.call({
+            path: `/atomicassets/v1/assets/${asset_id}/logs${queryParams}`,
+            method: 'GET',
+            responseType: Assets.ActionLogsResponse,
+        })
+    }
+
+    async get_collections(options?: {
+        author?: NameType[]
+        match?: string
+        authorized_account?: NameType
+        notify_account?: NameType
+        collection_blacklist?: NameType[]
+        collection_whitelist?: NameType[]
+        collection_name?: NameType[]
+        ids?: UInt64Type[]
+        lower_bound?: string
+        upper_bound?: string
+        before?: number
+        after?: number
+        page?: number
+        limit?: number
+        order?: 'asc' | 'desc'
+        sort?: 'created' | 'collection_name'
+    }) {
+        const queryParts = {}
+
+        for (const [key, value] of Object.entries(options || {})) {
+            queryParts[key] = value as string
```

Comment:
same here

Reply:
There are a quite a few of these so I'm not going to leave comments on all of them, but in general I think that we should only cast as a string if we are sure that the variable will be a string at runtime 😄